beacon-chain/sync/inclusion_list.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/prysmaticlabs/prysm/v5/time/slots"
+	"google.golang.org/protobuf/proto"
+)
+
+// validateInclusionList validates an incoming inclusion list message.
+// Returns appropriate validation results based on the following rules:
+// [REJECT] The slot `message.slot` is equal to the previous or current slot.
+// [IGNORE] The slot `message.slot` is equal to the current slot, or it is equal to the previous slot and the current time
+//
+//	is less than attestation_deadline seconds into the slot.
+//
+// [IGNORE] The inclusion_list_committee for slot `message.slot` on the current branch corresponds to `message.inclusion_list_committee_root`,
+//
+//	as determined by `hash_tree_root(inclusion_list_committee) == message.inclusion_list_committee_root`.
+//
+// [REJECT] The validator index `message.validator_index` is within the inclusion_list_committee corresponding to `message.inclusion_list_committee_root`.
+// [REJECT] The transactions `message.transactions` length is within the upper bound MAX_TRANSACTIONS_PER_INCLUSION_LIST.
+// [IGNORE] The message is either the first or second valid message received from the validator with index `message.validator_index`.
+// [REJECT] The signature of `inclusion_list.signature` is valid with respect to the validator index.
+func (s *Service) validateInclusionList(ctx context.Context, id peer.ID, msg *pubsub.Message) (pubsub.ValidationResult, error) {
+	// Skip self-published messages.
+	if id == s.cfg.p2p.PeerID() {
+		return pubsub.ValidationAccept, nil
+	}
+
+	// Ignore if the node is currently syncing.
+	if s.cfg.initialSync.Syncing() {
+		return pubsub.ValidationIgnore, nil
+	}
+
+	// Validate topic presence.
+	if msg.Topic == nil {
+		return pubsub.ValidationReject, errInvalidTopic
+	}
+
+	// Decode the pubsub message into the appropriate type.
+	m, err := s.decodePubsubMessage(msg)
+	if err != nil {
+		return pubsub.ValidationReject, err
+	}
+	il, ok := m.(*eth.SignedInclusionList)
+	if !ok {
+		return pubsub.ValidationReject, errWrongMessage
+	}
+
+	// Check for nil inclusion list.
+	if err := helpers.ValidateNilSignedInclusionList(il); err != nil {
+		return pubsub.ValidationIgnore, err
+	}
+
+	// Validate slot constraints.
+	currentSlot := s.cfg.clock.CurrentSlot()
+	if il.Message.Slot == currentSlot || il.Message.Slot+1 == currentSlot {
+		return pubsub.ValidationReject, errors.New("slot is equal to the previous or current slot")
+	}
+	secondsSinceSlotStart, err := slots.SecondsSinceSlotStart(currentSlot, uint64(s.cfg.chain.GenesisTime().Unix()), uint64(time.Now().Unix()))
+	if err != nil {
+		return pubsub.ValidationIgnore, err
+	}
+	deadline := params.BeaconConfig().SecondsPerSlot / params.BeaconConfig().IntervalsPerSlot
+	if il.Message.Slot+1 == currentSlot && secondsSinceSlotStart > deadline {
+		return pubsub.ValidationIgnore, errors.New("slot is equal to the current slot, or it is equal to the previous slot and the current time is less than attestation_deadline seconds into the slot")
```

Comment:
It seems the error messages here show why it went wrong instead of what it should be.